From 5a2537095621da25c8967a0ccdbb8b0029e14c94 Mon Sep 17 00:00:00 2001
From: Anson Huang <b20788@freescale.com>
Date: Fri, 18 Sep 2015 23:55:00 +0800
Subject: [PATCH] MLK-11566-5 drvier: imx_sema4: add dummy function

When CONFIG_IMX_SEMA4 is NOT set, some modules may
still call SEMA4 APIs, need to add dummy function
to make sure build pass.

Signed-off-by: Anson Huang <b20788@freescale.com>
---
 include/linux/imx_sema4.h | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/include/linux/imx_sema4.h b/include/linux/imx_sema4.h
index 2a259f6..50421c6 100644
--- a/include/linux/imx_sema4.h
+++ b/include/linux/imx_sema4.h
@@ -60,8 +60,27 @@ extern struct imx_sema4_mutex *mcc_shm_ptr;
 
 struct imx_sema4_mutex *
 	imx_sema4_mutex_create(u32 dev_num, u32 mutex_num);
+#ifdef CONFIG_IMX_SEMA4
 int imx_sema4_mutex_destroy(struct imx_sema4_mutex *mutex_ptr);
 int imx_sema4_mutex_trylock(struct imx_sema4_mutex *mutex_ptr);
 int imx_sema4_mutex_lock(struct imx_sema4_mutex *mutex_ptr);
 int imx_sema4_mutex_unlock(struct imx_sema4_mutex *mutex_ptr);
+#else
+static inline int imx_sema4_mutex_destroy(struct imx_sema4_mutex *mutex_ptr)
+{
+	return 0;
+}
+static inline int imx_sema4_mutex_trylock(struct imx_sema4_mutex *mutex_ptr)
+{
+	return 0;
+}
+static inline int imx_sema4_mutex_lock(struct imx_sema4_mutex *mutex_ptr)
+{
+	return 0;
+}
+static inline int imx_sema4_mutex_unlock(struct imx_sema4_mutex *mutex_ptr)
+{
+	return 0;
+}
+#endif
 #endif /* __LINUX_IMX_SEMA4_H__ */
-- 
1.8.0

